fix(route): shu 上海大学

don-fon opened the pull request:
<!-- 
如有疑问，请参考 https://docs.rsshub.app/joinus/new-rss/submit-route.html
Reference: https://docs.rsshub.app/en/joinus/new-rss/submit-route.html
-->

## 该 PR 相关 Issue / Involved Issue

Close #

## 路由地址示例 / Example for the Proposed Route(s)

<!--
请在 `routes` 区域填写以 / 开头的完整路由地址，否则你的 PR 将会被无条件关闭。
如果路由包含在文档中列出可以完全穷举的参数（例如分类），请依次全部列出。

Please include route starts with /, with all required and optional parameters. Fail to comply will result in your pull request being closed automatically.
```route
/some/route
/some/other/route
/dont/use/this/or/modify/it
/use/the/fenced/code/block/below
```
如果你的 PR 与路由无关, 请在 `routes` 区域 填写 `NOROUTE`，而不是直接删除 `routes` 区域。否则你的 PR 将会被无条件关闭。
If your changes are not related to route, please fill in `routes` section with `NOROUTE`. Fail to comply will result in your PR being closed.
-->

```routes
/shu/news
/shu/research
/shu/notice
/shu/jwc/notice
/shu/jwc/news
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [ ] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
网站改版

don-fon committed:
fix(route): shu 上海大学

TonyRL commented on lib/routes/universities/shu/jwc.js:
```diff
@@ -40,23 +41,26 @@ module.exports = async (ctx) => {
                 return Promise.resolve(JSON.parse(cache));
             }
+            const date = new Date(item.date);
+            const time_zone = 8;
+            const server_offset = date.getTimezoneOffset() / 60;
             const respond = await got.get(itemUrl);
             const $ = cheerio.load(respond.data);
             const single = {
                 title: item.title,
                 link: itemUrl,
-                author: $('.normal').next().text().trim().slice(0, -3),
+                author: $('[id$=_lblUser]').text().trim(),
+                pubDate: new Date(date.getTime() - 60 * 60 * 1000 * (time_zone + server_offset)).toUTCString(),
```
Please use timezone utility instead of reinventing the wheel.
docs: https://docs.rsshub.app/joinus/pub-date.html

don-fon committed:
Optimize code and update documentation

- shu/index.js & shu/jwc.js
    - use @/utils/parse-date to parse date
    - change "item.description" from text to HTML
    - change "url.resolve" (deprecated) to "new URL"
    - change the way of mapping "ctx.params.type" to link
        - now it accepts more route types
- university.md
    - update doc for new types
- router.js
    - add an alias because the website changed its host

TonyRL commented on lib/routes/universities/shu/jwc.js:
```diff
@@ -45,18 +37,18 @@ module.exports = async (ctx) => {
-        title: '上海大学教务处--' + title,
-        link: 'http://www.jwc.shu.edu.cn/',
+        title,
+        link: host,
```
```suggestion
        link,
```

TonyRL commented on docs/university.md:
```diff
@@ -2482,19 +2482,19 @@ jsjxy.hbut.edu.cn 证书链不全，自建 RSSHub 可设置环境变量 NODE_TLS
 
 ### 上海大学教务处通知公告
 
 <Route author="tuxinghuan" example="/shu/jwc/notice" path="/shu/jwc/:type?" :paramsDesc="['消息类型,默认为`notice`']">
```
```suggestion
<Route author="tuxinghuan" example="/shu/jwb/notice" path="/shu/jwb/:type?" :paramsDesc="['消息类型,默认为`notice`']">
```
https://github.com/DIYgod/RSSHub/blob/b33993a7a4e5175659da5b8ad221c37f27c49c01/lib/router.js#L749

don-fon committed:
Update lib/routes/universities/shu/index.js

Co-authored-by: Tony <TonyRL@users.noreply.github.com>

don-fon committed:
Update lib/routes/universities/shu/jwc.js

Co-authored-by: Tony <TonyRL@users.noreply.github.com>

don-fon committed:
Update docs/university.md

Co-authored-by: Tony <TonyRL@users.noreply.github.com>

TonyRL committed:
refactor: migrate to v2
